registry/handlers/referrers.go

```diff
@@ -76,8 +102,18 @@ func (h *referrersHandler) GetReferrers(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
-	if referrers == nil {
+	startIndex := pageNumber * pageSize
+
+	if referrers == nil || startIndex >= len(referrers) {
 		referrers = []v1.Descriptor{}
+	} else {
+		// if there's only 1 page of results left
+		if len(referrers)-startIndex <= pageSize {
+			referrers = referrers[startIndex:]
+		} else {
+			referrers = referrers[startIndex:(startIndex + pageSize)]
+			w.Header().Set("Link", generateLinkHeader(h.Repository.Named().Name(), h.Digest.String(), artifactTypeFilter, pageSize, pageNumber+1))
+		}
 	}
```

Comment:
nit: code style: readability
```suggestion
	startIndex := pageNumber * pageSize
	if referrers == nil || len(referrers) <= startIndex {
		referrers = []v1.Descriptor{}
	} else if len(referrers) <= startIndex+pageSize {
		// only 1 page of results left
		referrers = referrers[startIndex:]
	} else {
		referrers = referrers[startIndex:(startIndex + pageSize)]
		w.Header().Set("Link", generateLinkHeader(h.Repository.Named().Name(), h.Digest.String(), artifactTypeFilter, pageSize, pageNumber+1))
	}
```

Reply:
changed.